projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/DataJobsDeploymentController.java

```diff
@@ -111,7 +112,12 @@ public ResponseEntity<List<DataJobDeploymentStatus>> deploymentList(
       String teamName, String jobName, String deploymentId, DataJobMode dataJobMode) {
     // TODO: deploymentId and mode not implemented
     if (jobsService.jobWithTeamExists(jobName, teamName)) {
-      return deploymentAsList(jobName.toLowerCase());
+      var response = deploymentAsList(jobName.toLowerCase());
+      if (response.getStatusCodeValue() == 404) {
```

Comment:
I don't think this is a proper solution to the regression. You should rework the code https://github.com/vmware/versatile-data-kit/pull/2800.